plugin/trino-hive/src/test/java/io/trino/plugin/hive/metastore/glue/TestHiveGlueMetastoreCompatibility.java

```diff
@@ -0,0 +1,92 @@
+import com.google.common.collect.ImmutableMap;
+import io.trino.Session;
+import io.trino.plugin.hive.TestingHivePlugin;
+import io.trino.testing.AbstractTestQueryFramework;
+import io.trino.testing.DistributedQueryRunner;
+import io.trino.testing.QueryRunner;
+import io.trino.testing.sql.TestTable;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.Test;
+
+import java.io.IOException;
+import java.nio.file.Path;
+import java.util.Map;
+
+import static com.google.common.io.MoreFiles.deleteRecursively;
+import static com.google.common.io.RecursiveDeleteOption.ALLOW_INSECURE;
+import static io.trino.plugin.hive.metastore.glue.GlueHiveMetastore.createTestingGlueHiveMetastore;
+import static io.trino.testing.TestingNames.randomNameSuffix;
+import static io.trino.testing.TestingSession.testSessionBuilder;
+
+/*
+ * TestHiveGlueMetastoreCompatibility currently uses AWS Default Credential Provider Chain,
+ * See https://docs.aws.amazon.com/sdk-for-java/v1/developer-guide/credentials.html#credentials-default
+ * on ways to set your AWS credentials which will be needed to run this test.
+ */
+public class TestHiveGlueMetastoreCompatibility
+        extends AbstractTestQueryFramework
+{
+    protected static final String SCHEMA = "test_hive_glue_" + randomNameSuffix();
+
+    private String dataDirectory;
+    private GlueHiveMetastore glueMetastore;
+
+    @Override
+    protected QueryRunner createQueryRunner()
+            throws Exception
+    {
+        Session session = testSessionBuilder()
+                .setCatalog("hive")
+                .setSchema(SCHEMA)
+                .build();
+        DistributedQueryRunner queryRunner = DistributedQueryRunner.builder(session).build();
+
+        dataDirectory = queryRunner.getCoordinator().getBaseDataDir().resolve("hive_data").toString();
+        glueMetastore = createTestingGlueHiveMetastore(dataDirectory);
+
+        queryRunner.installPlugin(new TestingHivePlugin(glueMetastore));
+
+        Map<String, String> connectorProperties = ImmutableMap.<String, String>builder()
+                .put("hive.security", "allow-all")
+                .buildOrThrow();
+
+        queryRunner.createCatalog("hive", "hive", connectorProperties);
+        queryRunner.execute("CREATE SCHEMA " + SCHEMA);
+
+        return queryRunner;
+    }
+
+    @AfterClass(alwaysRun = true)
+    public void tearDown()
+            throws IOException
+    {
+        if (glueMetastore != null) {
+            glueMetastore.dropDatabase(SCHEMA, false);
+            deleteRecursively(Path.of(dataDirectory), ALLOW_INSECURE);
+        }
+    }
+
+    @Test
+    public void testSetColumnType()
+    {
+        try (TestTable table = new TestTable(getQueryRunner()::execute, "test_set_column_type_", "AS SELECT CAST(123 AS integer) AS col")) {
+            assertUpdate("ALTER TABLE " + table.getName() + " ALTER COLUMN col SET DATA TYPE bigint");
+            assertQuery("SELECT * FROM " + table.getName(), "VALUES 123");
```

Comment:
use `assertThat(query(`
then you will implicitly also verify the data comes indeed as bigint.